docs/sql-ref-datatypes.md

```diff
@@ -314,3 +314,49 @@ SELECT COUNT(*), c2 FROM test GROUP BY c2;
 |        3| Infinity|
 +---------+---------+
 ```
+
+#### Data type compatibility
+
+The following is the hierarchy of data type compatibility and the possible implicit conversions that can be made. In an operation involving different and compatible data types, these will be promoted to the lowest common top type to perform the operation.
```

Comment:
The current description looks ambiguous and many topics get mixed up, I think. What's the topics you would like to describe in this section? If you want to describe type conversion with the default mode (ansi=false), I think we need to pick up three categories: [explicit casting, type coercion, and store assignment casting](https://spark.apache.org/docs/latest/sql-ref-ansi-compliance.html#type-conversion). Anyway, we need a clearer structure to describe type behaviours for easy-to-read user documents.

Reply:
Thank you very much! I didn't know about Ansi property. I will redo the PR with this information and try to explain it with a better structure and also include a screenshot